state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2093,6 +2105,50 @@ pub mod pallet {
 				},
 			};
 		}
+
+		pub(super) fn take_refund_fee(
+			total_input_amount: AssetAmount,
+			input_asset: Asset,
+		) -> Result<AssetAmount, DispatchError> {
+			let refund_fee_usdc = MinimumNetworkFeePerChunk::<T>::get();
+
+			let required_refund_fee_as_input_asset = if input_asset == STABLE_ASSET {
+				refund_fee_usdc
+			} else {
+				with_transaction_unchecked(|| {
+					TransactionOutcome::Rollback(T::SwappingApi::swap_single_leg(
+						STABLE_ASSET,
+						input_asset,
+						refund_fee_usdc,
+					))
+				})?
+			};
+
+			let (remaining_amount_to_refund, refund_fee) =
+				if required_refund_fee_as_input_asset < total_input_amount {
+					(
+						total_input_amount - required_refund_fee_as_input_asset,
+						required_refund_fee_as_input_asset,
+					)
+				} else {
+					(0, total_input_amount)
+				};
+
+			if refund_fee > 0 {
+				Self::init_swap_request(
+					input_asset,
+					refund_fee,
+					Asset::Flip,
+					SwapRequestType::NetworkFee,
```

Comment:
It's strange that we allow the specification of a non-FLIP asset for NetworkFee swaps at the interface level? Not strictly this PR related, but wanted to point that out

Reply:
I was initially sceptical about making SwapRequestHandler more complex by adding more specialised methods (I think @dandanlen was suggesting something similar for on-chian swaps), but I'm now warming up to this idea, especially since we can provide a default implementation (for e.g `init_network_fee_swap_request`) that calls `init_swap_request` internally. Then it would be slightly easier to find where network fee is charged, and we can make sure that we convert into the right asset and that other parameters are set correctly (e.g. empty broker fees, SwapOrigin::Internal etc).